roles/matrix-bridge-appservice-slack/defaults/main.yml

```diff
@@ -0,0 +1,92 @@
+slack_control_room_id: ''
+matrix_appservice_slack_bot_name: 'slackbot'
+
+# Controls the SLACK_PORT and MATRIX_PORT of the installation
+matrix_appservice_slack_matrix_port: 8008
+matrix_appservice_slack_slack_port: 9003
+
+matrix_appservice_slack_homeserver_address: 'http://matrix-synapse:{{ matrix_appservice_slack_matrix_port }}'
+matrix_appservice_slack_homeserver_domain: '{{ matrix_domain }}'
+matrix_appservice_slack_appservice_address: 'http://matrix-appservice-slack:{{ matrix_appservice_slack_matrix_port }}'
+
+# A list of extra arguments to pass to the container
+matrix_appservice_slack_container_extra_arguments: []
+
+# List of systemd services that matrix-appservice-slack.service depends on.
+matrix_appservice_slack_systemd_required_services_list: ['docker.service']
+
+# List of systemd services that matrix-appservice-slack.service wants
+matrix_appservice_slack_systemd_wanted_services_list: []
+
+matrix_appservice_slack_appservice_token: ''
+matrix_appservice_slack_homeserver_token: ''
+matrix_appservice_slack_id_token: ''
+
+matrix_appservice_slack_configuration_yaml: |
+    slack_hook_port: {{ matrix_appservice_slack_slack_port }}
+    #{{matrix_homeserver_url}} had been replaced with {{matrix_domain}} hmmmm....
+    inbound_uri_prefix: "{{ matrix_homeserver_url }}:{{ matrix_appservice_slack_slack_port }}"
+    bot_username: "{{ matrix_appservice_slack_bot_name }}"
+    username_prefix: "slackuser_"
+
+    homeserver:
+        url: "{{ matrix_homeserver_url }}:{{matrix_appservice_slack_matrix_port}}"
+        server_name: "{{ matrix_domain }}"
+
+    matrix_admin_room: "{{ slack_control_room_id }}"
+
+
+matrix_appservice_slack_configuration_extension_yaml: |
+    #slack_hook_port: 9898
+    #inbound_uri_prefix: "https://my.server.here:9898/"
+    #bot_username: "slackbot"
+    #username_prefix: "slack_"
+    # Optional
+    #slack_master_token: "abc-123-def"
+    # Optional
+    #matrix_admin_room: "!aBcDeF:matrix.org"
+    #homeserver:
+    #    url: http://localhost:8008
+    #    server_name: my.server
+    # Optional
+    #tls:
+    #    key_file: /path/to/tls.key
+    #    crt_file: /path/to/tls.crt
+    #logging:
+    #    console: "info"
+    #    files:
+    #    - "./debug.log": "info"
+    #- "./error.log": "error"
+
+matrix_appservice_slack_configuration_extension: "{{ matrix_appservice_slack_configuration_extension_yaml|from_yaml if matrix_appservice_slack_configuration_extension_yaml|from_yaml else {} }}"
+
+matrix_appservice_slack_configuration: "{{ matrix_appservice_slack_configuration_yaml|from_yaml|combine(matrix_appservice_slack_configuration_extension, recursive=True) }}"
+
+matrix_appservice_slack_registration_yaml: |
+  id: "{{ matrix_appservice_slack_id_token }}"
+  as_token: "{{ matrix_appservice_slack_appservice_token }}"
+  hs_token: "{{ matrix_appservice_slack_homeserver_token }}"
+  namespaces:
+    users:
+    - exclusive: true
+      regex: '@slack_.*'
```

Comment:
I do wonder if the regex should be so generic (everything prefixed with `@slack_`), or if it should be more strict (mentioning the Matrix domain, etc.)

Reply:
This is something that the user should be able to customize, so I'll add a variable for it.

Reply:
For other bridges we do the obvious and safe thing (only allowing bridging for users on this homeserver). If people want to change it, we just ask them to override the whole registration configuration (e.g. `matrix_appservice_slack_registration_yaml`).

However, adding a variable sounds better and makes overriding easier!